### PR TITLE
Refactor deploy workflows

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
#### Summary of Changes (What does this PR do?)
- Uses Secret Manager secrets as source of environmental variables
- Cleans up workflows 
- Fixes a docs CICD pipeline typo

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* You can only do static code analysis

#### Screenshots (optional)

### CODE REVIEW

**Comment by 123MwanjeMike** [comment]
Duplicate

File: .github/workflows/deploy-frontend-to-preview-env.yml [hunk 1]
```diff
@@ -88,42 +88,33 @@ jobs:
...
 
-      - name: NPM Setup and Build
+      - name: npm Install and Build
         uses: actions/setup-node@v2
         with:
           node-version: "12"
       - run: |
           cd netmanager/
-
-          export REACT_APP_BASE_URL=${{ secrets.REACT_APP_BASE_URL_STAGE }}
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-frontend-to-preview-env.yml:115]: These values are now being picked directly from secret manager and added to a .env file

File: .github/workflows/deploy-frontends-to-production.yml [hunk 2]
```diff
@@ -377,22 +353,22 @@ jobs:
...
+          method: kubeconfig
+          kubeconfig: ${{ secrets.K8S_CONFIG_PROD }}
 
       - name: Deploy to K8S
         run: |
-          kubectl apply -f k8s/docs/prod-airqo-next-platform.yaml
+          kubectl apply -f k8s/docs/prod-airqo-docs.yaml
           kubectl set image deployment/airqo-prod-docs airqo-prod-docs=${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-prod-docs:${{ needs.image-tag.outputs.build_id }} -n production
-          kubectl annotate deployment/aairqo-prod-docs kubernetes.io/change-cause="Image updated to ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-prod-docs:${{ needs.image-tag.outputs.build_id }} on ${{ needs.image-tag.outputs.datetime }}" -n production
+          kubectl annotate deployment/airqo-prod-docs kubernetes.io/change-cause="Image updated to ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-prod-docs:${{ needs.image-tag.outputs.build_id }} on ${{ needs.image-tag.outputs.datetime }}" -n production
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-frontends-to-production.yml:374]: [correct docs CICD typos](https://github.com/airqo-platform/AirQo-frontend/pull/838/commits/cc5a627426999b03548ece541d44f62e412c204b)

File: .github/workflows/deploy-frontends-to-production.yml [hunk 3]
```diff
@@ -93,62 +93,52 @@ jobs:
...
-      - name: Login to K8S
-        uses: azure/k8s-set-context@v1
-        with:
-          method: kubeconfig
-          kubeconfig: ${{ secrets.K8S_CONFIG_PROD }}
+      - name: Setup Cloud SDK
+        uses: google-github-actions/setup-gcloud@v0
+
+      - name: Create .env file
+        run: gcloud secrets versions access latest --secret="prod-env-netmanager" > netmanager/.env
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-frontends-to-production.yml:105]: Creating the .env file in the specific application folder

File: .github/workflows/deploy-frontends-to-production.yml [hunk 4]
```diff
@@ -93,62 +93,52 @@ jobs:
...
-        run: |
-          cd netmanager/
-          gcloud secrets versions access latest --secret="prod-env-netmanager" > .env
-          kubectl create configmap --dry-run=client -o yaml --from-env-file=.env env-platform-production | kubectl replace -f - -n production
+          method: kubeconfig
+          kubeconfig: ${{ secrets.K8S_CONFIG_PROD }}
 
       - name: Deploy to K8S
         run: |
+          echo "=============== Update the corresponding configmap ==============="
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-frontends-to-production.yml:139]: Move the "update corresponding configmap" step into the k8s step

File: .github/workflows/run-tests.yml [hunk 5]
```diff
@@ -48,12 +48,18 @@ jobs:
...
+        uses: google-github-actions/setup-gcloud@v0
+
       - name: Create .env file
         run: |
-          cd platform/
-          echo "========== Creating app.yaml file =========="  
-          echo "NEXT_PUBLIC_ORIGINAL_PLATFORM=${{ secrets.NEW_PLATFORM_STAGE_ORIGINAL_PLATFORM }}" > .env
-          echo "INSTRUMENT_CODE=1" >> .env
+          gcloud secrets versions access latest --secret="sta-env-next-platform" > platform/.env
+          echo "INSTRUMENT_CODE=1" >> platform/.env
```
Comment (123MwanjeMike) [inline .github/workflows/run-tests.yml:62]: Append test-specific environment variable to .env

**Review by Baalmart — COMMENTED** [review]
Thanks @123MwanjeMike , this looks okay. I will let @NoahNsimbe take a look too.

**Review by NoahNsimbe — APPROVED** [review]
LGTM

**Review by Baalmart — APPROVED** [review]
LGTM

**Pull request merged**